src/chttpd/src/chttpd_db.erl

```diff
@@ -1960,6 +1962,13 @@ validate_security_can_be_edited(DbName) ->
         {_,_} -> ok
     end.
 
+validate_revs(Doc, NewEdits) ->
```

Comment:
It might be simpler to check a few things directly in the function head

```
validate_revs(_Doc, true) ->
    ok
validate_revs(#doc{revs = {0, []}, false) ->
    throw({bad_request, ....})
validate_revs(_Doc, false) ->
     ok